### PR TITLE
chore: test solution 

### PR DESCRIPTION
**Pull request by abilevych**
# What :computer:

- created an infrastructure for tests
- optimized the test code related to Hello World scenario
- integrated the CI/CD test execution per commit with the era-test-node

# Why :hand:

- integration test checks per PR

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -7,23 +7,6 @@
   resolved "https://registry.yarnpkg.com/@aashutoshrathi/word-wrap/-/word-wrap-1.2.6.tgz#bd9154aec9983f77b3a034ecaa015c2e4201f6cf"
   integrity sha512-1Yjs2SvM8TflER/OD3cOjhWWOZb58A2t7wpE2S9XfBYTiIl+XFhQG2bjy4Pu1I+EAlCNUzRDYDdFwFYUKvXcIA==
 
-"@achrinza/node-ipc@9.2.2":
```
Comment (idea404) [inline yarn.lock:10]: How come the lockfile changed if there are no changes in the package.json deps?

File: hello-world/test/main.test.ts [hunk 2]
```diff
@@ -1,31 +1,39 @@
...
 
-    const setGreetingTx = await greeter.setGreeting('Hola, mundo!');
-    // wait until the transaction is mined
-    await setGreetingTx.wait();
+    result = await contract.greet();
 
-    expect(await greeter.greet()).to.equal('Hola, mundo!');
+    expect(result).to.equal("Hola, mundo!");
   });
 });
```
Comment (idea404) [inline hello-world/test/main.test.ts:39]: I'd try adding some failing tests too, like passing wrong param types for example. Important to demonstrate that testing also involves testing unexpected scenarios

**Review by idea404 — CHANGES_REQUESTED** [review]
Some remarks, a good starting point

File: tests/README.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+![Gradient Banner](https://github.com/matter-labs/tutorials/assets/10233439/8efffb9b-ad1f-4bf2-8f73-9cab8f7ccd22)
+
+# zkSync Tutorials
+
+This document contains an instruction of the test infrastructure preparation/running tests locally.
+
+## The environment preparation
+
+To have a possibility of test execution there needs to install a local node (L2) based on the next repository https://github.com/matter-labs/era-test-node.
```
Comment (idea404) [inline tests/README.md:9]:
```suggestion
To execute tests, you need to install a local node (L2) from the following repository: https://github.com/matter-labs/era-test-node
```

File: tests/README.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
+![Gradient Banner](https://github.com/matter-labs/tutorials/assets/10233439/8efffb9b-ad1f-4bf2-8f73-9cab8f7ccd22)
+
+# zkSync Tutorials
+
+This document contains an instruction of the test infrastructure preparation/running tests locally.
```
Comment (idea404) [inline tests/README.md:5]:
```suggestion
This document provides instructions for preparing the test infrastructure and running tests locally.
```

**Review by idea404 — CHANGES_REQUESTED** [review]
Some minor suggestions

**Review by idea404 — APPROVED** [review]

File: tests/README.md (hunk 3 above)
Comment (AMER88795) [reply tests/README.md]: Yas[]()[]()https://zerion.io/premium

File: tests/README.md (hunk 4 above)
Comment (AMER88795) [reply tests/README.md]: googleads_mobile_ios_swift_adaptive_banner_example.zip

File: tests/README.md (hunk 3 above)
Comment (AMER88795) [reply tests/README.md]: @

**Pull request merged**